### DISCUSSION TITLE
Is there any way to determine, at the level of grammar, that the order of expressions does not matter?

### DISCUSSION THREAD

**Discussion by kafkiansky** [pest-parser/pest#842] [Q&A]
A simple case: I have a set of blocks, the order of which is not important. In example
```pest
WHITESPACE = _{ " " | "\t" | "\r" | "\n" }
a = {"a"}
b = {"b"}
```
I want the user to be able to write both `a b` and `b a`, with both `a` and `b` being required, but their order is not important. The workaround is:
```pest
or = { (a ~ b) | (b ~ a) }
```

But I have too many blocks for which describing such a thing would be difficult.

**Comment by phoenixr-codes** [comment]
This **would** be possible with a `PEEK_ANY` keyword.

From my understanding `PEEK_ALL` matches the first item in the stack, then the second, ….

So it probably would make sense to invent a `PEEK_ANY` keyword?

A solution with this could look like this:

```haskell
blocks = {
    "a"
  | "b"
  | "c"
  | "d"
}

main = {
  PUSH(blocks)
  ~ (
    !PEEK_ANY
    ~ PUSH(blocks)
  ){3}
}
```

After all I think the workaround is the only solution that works for this case now.
